App/BitBar/ExecutablePlugin.m

```diff
@@ -70,86 +70,97 @@ -(void)performRefreshNow:(NSMenuItem*)menuItem{
 }
 
 -(BOOL)refresh {
+  __unsafe_unretained ExecutablePlugin *weakSelf = self;
   [self.lineCycleTimer invalidate];
   self.lineCycleTimer = nil;
   [self.refreshTimer invalidate];
   self.refreshTimer = nil;
-    
+
   // execute command
   dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT,0),  ^{
-    [self refreshContentByExecutingCommand];
+    [weakSelf refreshContentByExecutingCommand];
     dispatch_sync(dispatch_get_main_queue(), ^{
-      
-      self.lastUpdated = NSDate.new;
-      
-      [self rebuildMenuForStatusItem:self.statusItem];
-      
-      // reset the current line
-      self.currentLine = -1;
-      
-      // update the status item
-      [self cycleLines];
-      
-      // sort out multi-line cycler
-      if (self.isMultiline) {
+      if (weakSelf) {
+        __strong ExecutablePlugin* strongSelf = weakSelf;
```

Comment:
Actually just changed it to a `dispatch_async` because I could, no real effect.  No errors in Console or anything weird...  So unless @matryer knows something I don't, it's probably a safe change.